dash_labs/dashdown.py

```diff
@@ -0,0 +1,419 @@
+           If `True`, code blocks will be executed.  This may also be set within the code block with the comment
+            # exec-code-true or # exec-code-false
+
+        - `side_by_side` (boolean; default False):
+          If `True`, the code block will be displayed on the left and the app output on the right on large screens.
+          If `False`, or on small screens, code block will be displayed on top and the output will be on the bottom.
+          This may also be set within the code block with the comment # side-by-side-true or # side-by-side-false.
+
+        - `code_first` (boolean; default True):
+          If `True`, the code block will be displayed on the top and output on the bottom (or on the left if side by side).
+          This may also be set within the code block with the comment # code-first-true or # code-first-false
+
+        - `code_markdown_props`(dict; default ):  A dictionary of properties passed into the dcc.Markdown component that
+           displays the code blocks. Does not accept user-supplied `id`, `children`, or `dangerously_allow_html` props.
+           This may also be set within a code block with the comment # code-markdown-props-{...}
+
+        - `text_markdown_props`(dict; default ):  A dictionary of properties passed into the dcc.Markdown component that
+           displays the Markdown text other than code blocks. Does not accept user-supplied `id`, `children` props.
+
+        - `clipboard_props`(dict; default ):  A dictionary of properties passed into the dcc.Clipboard component. Does
+            not accept user-supplied `id`, `content`, 'target_id` props.
+            This may also be set within a code block with the comment # clipboard-props-{...}
+
+        - `app_div_props`(dict; default ):  A dictionary of properties passed into the html.Div component that contains
+          the output of the executed code blocks.  Does not accept user-supplied `id`, `children` props.
+          This may also be set within a code block with the comment # app-div-props-{...}
+
+        """
+        if dash_scope:
+            scope = dict(
+                dcc=dcc,
+                html=html,
+                Input=Input,
+                Output=Output,
+                State=State,
+                dash_table=dash_table,
+                px=px,
+                plotly=plotly,
+                dbc=dbc,
+                **(scope or {}),
+            )
+        elif not scope:
+            scope = {}
+
+        try:
+            with open(filename, "r") as f:
+                notebook = f.read()
+        except IOError as error:
+            warnings.warn(f"{error} supplied to MarkdownAIO", stacklevel=2)
+            return ""
+
+        # remove frontmatter which is delimited with 3 dashes
+        split_frontmatter = re.split(r"(^---[\s\S]*?\n*---)", notebook)
+        notebook = split_frontmatter[-1]
+        template = Template(notebook)
+        notebook = template.render(**(template_variables or {}))
```

Comment:
have you tried including another file with jinja in here? `{% includes code.md %}` or w/e the jinja syntax is?

curious if that works.

also curious what the path convention is here. do you need to do `{% includes pages/code.md %}` or can you just do `{% code.md %}`?

Reply:
This isn't necessary to use `{% includes code.md %}`.  Can use `MarkdownAIO("code.md")`
The path convention will be the same as the discussion above